extensions/2.0/Vendor/EXT_mesh_features/README.md

```diff
@@ -127,263 +138,466 @@ For example
 * Variable-length arrays are not supported.
-
-Additionally, the data type and bit depth of the image must be compatible with the property type. An 8-bit per pixel RGB image is only compatible with `UINT8` or normalized `UINT8` properties, and array properties thereof with three components or less. Likewise, a floating point property requires a floating point-compatible image format like KTX2 which may require additional extensions.
-
-Property textures are defined with the following steps:
-
-1. A class is defined in the root `EXT_mesh_features` extension object. This is used to describe the properties in the texture.
-2. A property texture is defined in the root `EXT_mesh_features.propertyTextures` object. This must reference the class ID defined in step 1.
-3. A property texture is associated with a primitive by listing the property texture ID in the `primitive.EXT_mesh_features.propertyTextures` array.
-<img src="figures/feature-texture.png"  alt="Property Texture" width="500">
-
-_Class and property texture_
-
-```jsonc
-{
-  "extensions": {
-    "EXT_mesh_features": {
-      "schema": {
-        "classes": {
-          "heatSample": {
-            "properties": {
-              "heatLoss": {
-                "type": "UINT8",
-                "normalized": true
-              },
-              "insulation": {
-                "type": "UINT8",
-                "normalized": true
-              },
-            }
-          }
-        }
-      },
-      "propertyTextures": [{
-        "class": "heatSample",
-        "index": 0,
-        "texCoord": 0,
-        "properties": {
-          "heatLoss": [0],
-          "insulation": [1]
-        }
-      }]
-    }
-  }
-}
-```
-
-_Primitive_
-
-```jsonc
-{
-  "primitives": [
-    {
-      "attributes": {
-        "POSITION": 0,
-        "TEXCOORD_0": 1
-      },
-      "indices": 2,
-      "material": 0,
-      "extensions": {
-        "EXT_mesh_features": {
-          "propertyTextures": [0]
-        }
-      }
-    }
-  ]
-}
-```
-
-
-A `propertyTexture` object extends the glTF [`textureInfo`](../../../../../specification/2.0/schema/textureInfo.schema.json) object. `texCoord` refers to the texture coordinate set of the referring primitive. The `properties` map specifies the texture channels providing data for all required class properties, and perhaps optional class properties. An array of integer index values identify channels, where multiple channels may be used only for fixed-length arrays of 2, 3, or 4 components. Channels of an `RGBA` texture are numbered 0–3 respectively, although specialized texture formats may allow additional channels. All values are stored in linear space.
+* Data type and bit depth of the image must be compatible with the property type.
+
+Consequently, an 8-bit per pixel RGB image is only compatible with `UINT8` or normalized `UINT8` properties, and array properties thereof with three components or fewer.
```

Comment:
okay, I think I'm okay if we permit these, but the wording in this paragraph should be tweaked so we don't imply that it must be a `UINT8` or an array.

Oh also these properties could potentially be signed integers, not just UINT8? sure, takes one more step when decoding, but it certainly fits in the texture.

Reply:
Hm I hadn't thought of signed integer textures. Is it an extra step, or just `isampler` instead of `usampler`? Haven't really worked with those before.

Maybe saying `only compatible with "UINT8" or normalized "UINT8"` is just too prescriptive, by trying to list all valid options for PNG and JPEG? E.g. it seems to prohibit enum properties and I don't know if that's the intention. How about:

> For example, an 8-bit per pixel RGB image may only contain `SINGLE`, fixed-length `ARRAY` (length ≤3), `VEC2`, or `VEC3` values composed of 8-bit component types.

Reply:
Done ✅